azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySets.java

```diff
@@ -3,25 +3,19 @@
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsDeletingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsListingByGroup;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsCreating;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsDeleting;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsListing;
+import com.microsoft.azure.management.resources.fluentcore.collection.ResourcesInGroup;
 
 /**
  * Entry point to availability set management API.
  */
-public interface AvailabilitySets extends
-        SupportsListing<AvailabilitySet>,
+public interface AvailabilitySets extends 
+        ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionBlank>,
         SupportsListingByGroup<AvailabilitySet>,
         SupportsGettingByGroup<AvailabilitySet>,
-        SupportsCreating<AvailabilitySet.DefinitionBlank>,
-        SupportsDeleting,
         SupportsDeletingByGroup {
-    /**
-     * Entry point to availability set management API within a specific resource group.
-     */
-    interface InGroup extends
-            SupportsListing<AvailabilitySet>,
-            SupportsCreating<AvailabilitySet.DefinitionCreatable>,
-            SupportsDeleting {}
+    
+    
+    public interface InGroup 
+        extends ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionAfterGroup> {
```

Comment:
I think a better solution is to have something like a base InGroup interface and everybody extends that.
It will look like
```java
// AfterGroupBase.java
public interface AfterGroupBase {
}
// ResourceInGroup.java
public interface ResourceInGroup<T, T extends AfterGroupBase> {
}
// AvailabilitySets.java
interface InGroup extends ResourceInGroup<AvailabilitySet, AvailabilitySet.DefinitionWithFoo> {
}
// AvailabilitySet.java
interface DefinitionWithGroup extends GroupableResource.DefinitionWithGroup<DefinitionWithFoo> {
}
interface DefinitionWithFoo extends AfterGroupBase {
}
```
So we reduce a significant amount of `AfterGroup` interfaces and when we move the `extends AfterGroupBase` flag from `DefinitionCreatable` to `DefinitionWithFoo` if we forget to change the `InGroup` interface, the compiler will yell.